internal/pkg/routes/mfx/mfx.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mfx
+
+import (
+	"fmt"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/export/internal/pkg/routes"
+	"github.com/mainflux/mainflux"
+	"github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/senml"
+	"github.com/nats-io/nats.go"
+)
+
+const (
+	// JSON represents SenML in JSON format content type.
+	JSON = "application/senml+json"
+
+	// CBOR represents SenML in CBOR format content type.
+	CBOR = "application/senml+cbor"
+)
+
+var formats = map[string]senml.Format{
+	JSON: senml.JSON,
+	CBOR: senml.CBOR,
+}
+
+type mfxRoute struct {
```

Comment:
`mfxRoute` wich containes `Route` does not look like a good solution to me.

Please define interface-only in the package, and here in the implementation define `Route` struct.

Take look here: https://github.com/mainflux/mproxy/blob/master/examples/simple/simple.go for a more elegant interface implementation.

Reply:
Thing that may look less obvious now since plain.go implemntation is removed is to have different implementation of Route interface but share common struct fields
So  routes.Route is default implementation of RouteIF interface and it is used just to send plain byte payload receieve on nats to mqtt.
mfx is different implementation it does some processing of payload received on nats, however it should contain same fields

Reply:
Do we need the `Publish` method at all and, if yes, do we need it exported?